lib/galaxy/webapps/galaxy/controllers/dataset.py

```diff
@@ -232,11 +232,17 @@ def transfer_status(self, trans, dataset_id, filename=None):
             return trans.app.object_store.file_ready(data.dataset)
 
     @web.expose
-    def display(self, trans, dataset_id=None, preview=False, filename=None, to_ext=None, chunk=None, **kwd):
+    def display(self, trans, dataset_id=None, preview=False, filename=None, to_ext=None, offset=None, ck_size=None, **kwd):
         data = self._check_dataset(trans, dataset_id)
         if not isinstance( data, trans.app.model.DatasetInstance ):
             return data
-        return data.datatype.display_data(trans, data, preview, filename, to_ext, chunk, **kwd)
+        # Ensure offset is an integer before passing through to datatypes.
+        if offset:
+            offset = int(offset)
+        # Ensure ck_size is an integer before passing through to datatypes.
+        if ck_size:
+            ck_size = int(ck_size)
+        return data.datatype.display_data(trans, data, preview, filename, to_ext, offset, ck_size, **kwd)
```

Comment:
Thank you! My intention was not to inject a rewrite in the bugfix, but I fear that without the removal of the unusued conflicting params it can break more things than it fixes! If you think that's too much for 16.04 we can target `dev`, either is fine for me.